BedrockServer.cpp

```diff
@@ -991,11 +991,20 @@ void BedrockServer::worker(int threadId)
                         // Roll back the transaction, it'll get re-run in the sync thread.
                         core.rollback();
 
-                        // We're not handling a writable command anymore.
-                        SINFO("Sending non-parallel command " << command->request.methodLine
-                              << " to sync thread. Sync thread has " << _syncNodeQueuedCommands.size()
-                              << " queued commands.");
-                        _syncNodeQueuedCommands.push(move(command));
+                        if (state == SQLiteNode::LEADING) {
+                            SINFO("Sending " << command->request.methodLine << " to sync thread.");
+                            _syncNodeQueuedCommands.push(move(command));
+                        } else if (state == SQLiteNode::STANDINGDOWN) {
+                            SINFO("Need to process command " << command->request.methodLine << " but STANDINGDOWN, moving to _standDownQueue.");
+                            _standDownQueue.push(move(command));
+                        } else if (!_clusterMessenger.sendToLeader(*command)) {
+                            // TODO: Also handle this better?
```

Comment:
What does this mean?

Reply:
Right now, if we fail to get a connection to leader, we just requeue the command and try again in a second, but that's pretty naive. Depending on why we fail to get a connection, we could probably do something better than this looping on the same commands until they work correctly.